casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -103,6 +94,66 @@ object Genesis {
                      )
     } yield genesisBlock
 
+  def getWallets[F[_]: Sync: Log: RaiseIOError](
+      maybeWalletsPath: Option[String],
+      defaultWalletPath: Path
+  ): F[Seq[Vault]] = {
+    def walletFromFile(walletsPath: Path): F[Seq[Vault]] =
+      for {
+        maybeLines <- SourceIO.open(walletsPath).use(_.getLines).attempt
+        wallets <- maybeLines match {
+                    case Right(lines) =>
+                      lines
+                        .traverse(fromLine(_) match {
+                          case Right(wallet) => wallet.some.pure[F]
+                          case Left(errMsg) =>
+                            Log[F]
+                              .warn(s"Error in parsing wallets file: $errMsg")
+                              .map(_ => none[Vault])
+                        })
+                        .map(_.flatten)
+                    case Left(ex) =>
+                      Log[F]
+                        .warn(
+                          s"Failed to read ${walletsPath.toAbsolutePath} for reason: ${ex.getMessage}"
+                        )
+                        .map(_ => Seq.empty[Vault])
+                  }
+      } yield wallets
+
+    maybeWalletsPath match {
+      case Some(walletsPathStr) =>
+        val walletsPath = Paths.get(walletsPathStr)
+        Monad[F].ifM(exists(walletsPath))(
+          walletFromFile(walletsPath),
+          Sync[F].raiseError(new Exception(s"Specified wallets file $walletsPath does not exist"))
+        )
+      case None =>
+        Monad[F].ifM(exists(defaultWalletPath))(
+          Log[F].info(s"Using default file $defaultWalletPath") >> walletFromFile(
+            defaultWalletPath
+          ),
+          Log[F]
+            .warn(
+              "No wallets file specified and no default file found. No wallets will exist at genesis."
+            )
+            .map(_ => Seq.empty[Vault])
+        )
+    }
+  }
+
+  def fromLine(line: String): Either[String, Vault] = line.split(",").filter(_.nonEmpty) match {
```

Comment:
if I understand correctly ```filter(_.nonEmpty)``` will silently accept empty fields in the wallets.txt 

I think this is a case where we should fail instead of silently going forward